### PR TITLE
Externalize timezone for execution windows as config property tasks.e…

### PR DESCRIPTION
**Pull request by duftler**
…xecutionWindow.timezone.

Replace reliance on jvm-wide setting of default timezone with Calendar instance.

### CODE REVIEW

**Comment by duftler** [comment]
@cfieber, @ajordens or @anotherchrisberry please review.

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/RestrictExecutionDuringTimeWindow.groovy [hunk 1]
```diff
@@ -52,24 +58,20 @@ class RestrictExecutionDuringTimeWindow extends LinearStage {
...
     TaskResult execute(Stage stage) {
-      TimeZone.'default' = TimeZone.getTimeZone("America/Los_Angeles")
-      System.setProperty("user.timezone", "America/Los_Angeles")
-
-      Date now = getCurrentDate()
+      Date now = new Date()
       Date scheduledTime
       try {
-        scheduledTime = getTimeInWindow(stage, getCurrentDate())
+        scheduledTime = getTimeInWindow(stage, new Date())
```
Comment (cfieber) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/RestrictExecutionDuringTimeWindow.groovy]: `now` instead of `new Date()`

Comment (duftler) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/RestrictExecutionDuringTimeWindow.groovy]: The passed-in `scheduledTime` param gets mutated within `getTimeInWindow()`. I tried to keep the changes minimal, and it seems the in/out parameter is mainly to facilitate testing.

Comment (cfieber) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/RestrictExecutionDuringTimeWindow.groovy]: I don't see the mutation in there (I see the scheduledTime.time assigned into a calendar). No big deal if I'm missing something there

**Comment by cfieber** [comment]
LGTM

**Comment by duftler** [comment]
Ah, you're right. I forgot I changed it to modify the calendar instead of the date. I'll make the change.

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/RestrictExecutionDuringTimeWindow.groovy (hunk 1 above)
Comment (duftler) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/RestrictExecutionDuringTimeWindow.groovy]: Done.

**Pull request merged**